apps/dashboard/src/main/java/com/akto/listener/InitializerListener.java

```diff
@@ -1053,14 +1058,36 @@ public static void updateTestEditorTemplatesFromGithub() {
                     loggerMaker.errorAndAddToDb(String.format("Error parsing yaml template file %s %s", template.getName(), e.toString()), LogDb.DASHBOARD);
                 }
 
-
                 if (testConfig == null) {
                     loggerMaker.errorAndAddToDb(String.format("Error parsing yaml template file %s", template.getName()), LogDb.DASHBOARD);
                 } else {
+                    Metadata metadata = testConfig.getMetadata();
+
+                    // Get deployment type and the appropriate template version
+                    String templateMinimumAktoVersion = null;
+
+                    if (DashboardMode.isLocalDeployment()) {
+                        templateMinimumAktoVersion = metadata.getMinAktoVersion();
+                    } else {
+                        templateMinimumAktoVersion = metadata.getMinOnpremVersion(); 
+                    }
+
+                    // Get dashboard version
+                    String dashboardVersion = DashboardVersion.getDashboardVersion();
+
+                    // Check if updated template is not supported by dashboard
+                    if (DashboardVersion.isSemanticVersionString(dashboardVersion)) {
```

Comment:
so basically if tag is not semantic(local/stage deployments), then we will always update, right? since if check will get skipped

Reply:
Yes